Makefile

```diff
@@ -0,0 +1,20 @@
+DOCKER := $(shell which docker)
+DOCKER_BUF := $(DOCKER) run --rm -v $(CURDIR):/workspace --workdir /workspace bufbuild/buf
+
+## proto-gen-docker: Generate protobuf files. Requires docker.
+proto-gen-docker:
+	@echo "--> Generating Protobuf files"
+	$(DOCKER_BUF) generate
+.PHONY: proto-gen-docker
+
+## proto-gen: Generate protobuf files. Requires buf.
+proto-gen:
+	@echo "--> Generating Protobuf files"
+	@buf generate
+.PHONY: proto-gen
```

Comment:
[question] do we want to support two different ways to generate the Protobuf files? I think it easier to maintain just one option and if that option is used in CI as well as locally. Since the `make proto-gen` command depends on a locally installed Buf CLI, preference for the Docker approach.

Ideally the Docker approach uses the same version of Buf locally and in CI.

Reply:
We could probably look to enforce it somehow. The docker image also isn't versioned (I actually had a problem where it wasn't working because I had an old image.

I think some people don't like using docker, hence why I added the other option, but it's simple enough that it probably doesn't need a make command so I will remove it

Reply:
Yea I think the Docker approach should also be versioned.